modules/light-clients/09-localhost/client_state.go

```diff
@@ -153,8 +169,10 @@ func (cs ClientState) UpdateStateOnMisbehaviour(_ sdk.Context, _ codec.BinaryCod
 func (cs ClientState) UpdateState(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, clientMsg exported.ClientMessage) []exported.Height {
 	height := clienttypes.GetSelfHeight(ctx)
 
-	clientState := NewClientState(ctx.ChainID(), height)
-	clientStore.Set([]byte(host.KeyClientState), clienttypes.MustMarshalClientState(cdc, clientState))
+	cs.ChainId = ctx.ChainID()
+	cs.LatestHeight = height
+
+	clientStore.Set([]byte(host.KeyClientState), clienttypes.MustMarshalClientState(cdc, &cs))
```

Comment:
just to make sure I understand the intention correctly, here we modify `cs` the method receiver, but it is not a pointer, so the CientState in the store will be updated, but the instance we act on i.e. 

```go
clientState.UpdateState(...)
```

will remain unchanged.

Reply:
yeah, correct, this is essentially the same as how the solomachine client state is doing updates also.